src/Nethermind/Nethermind.Consensus/Producers/PayloadAttributes.cs

```diff
@@ -54,7 +59,9 @@ public static class PayloadAttributesExtensions
     public static string ComputePayloadId(this PayloadAttributes payloadAttributes, BlockHeader parentHeader)
     {
         bool hasWithdrawals = payloadAttributes.Withdrawals is not null;
-        Span<byte> inputSpan = stackalloc byte[32 + 32 + 32 + 20 + (hasWithdrawals ? 32 : 0)];
+        bool hasBeaconStateRoot = payloadAttributes.BeaconStateRoot is not null;
+
+        Span<byte> inputSpan = stackalloc byte[32 + 32 + 32 + 20 + (hasWithdrawals ? 32 : 0) + (hasBeaconStateRoot ? 32 : 0)];
```

Comment:
Is this allocated space always completely written to?

If so adding `[SkipLocalsInit]` to method would help. If it relies on being auto zero'd and only partially writes then don't do that 😀

Reply:
Btw this is being refactored in `optimism` branch!

Reply:
well adding ``[SkipLocalsInit]`` breaks some tests